src/python/pants/invalidation/build_invalidator.py

```diff
@@ -58,7 +58,9 @@ def __init__(self, cache_key_gen_version=None):
     cache_key_gen_version - If provided, added to all cache keys. Allows you to invalidate
       all cache keys in a single pants repo, by changing this value in config.
     """
-
+    # TODO: Qualify the BuildInvalidator root with the cache_key_gen_version, instead of mixing
+    # it into the key, for uniformity with other qualifiers like the fprint strategy and the
```

Comment:
By "cache key" I mean the keys into the build invalidator. Pretend for the purpose of this discussion that there's no such thing as the artifact cache...  Anyway, my comment was an early statement of the problem I restated more explicitly above.